### PR TITLE
feat(swagger): Added gin-swagger integration and initial docs

### PR DESCRIPTION
**Pull request by theodesp**
## Description

This commit adds Swagger support by including [gin-swagger](https://github.com/swaggo/gin-swagger) and [swag](https://github.com/swaggo/swag) to the project. It then provides API Documentation for the Auth Login Endpoint. 

I chose to provide only support for Swagger in this PR and have subsequent tasks for completing the rest of the API Documentation endpoints. This is to make the whole process easier to review and approve. I will create another Issue outlining the checklist of available endpoints to document so that the work is broken up in manageable parts.

## Fixes
https://github.com/zinclabs/zinc/issues/156

## Screenshots
Screenshot 2022-05-31 at 10 09 26

### CODE REVIEW

File: pkg/handlers/auth/login.go [hunk 1]
```diff
@@ -21,27 +21,53 @@ import (
...
-	loggedInUser, validationResult := auth.VerifyCredentials(user.ID, user.Password)
-	resUser := gin.H{}
+	loggedInUser, validationResult := auth.VerifyCredentials(loginInput.ID, loginInput.Password)
+	var resUser LoginUser
 	if validationResult {
-		resUser = gin.H{
-			"_id":  loggedInUser.ID,
-			"name": loggedInUser.Name,
-			"role": loggedInUser.Role,
+		resUser = LoginUser{
```
Comment (theodesp) [inline pkg/handlers/auth/login.go:43]: Replaces gin Interface with typed Response Models.

File: docs/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+basePath: /api
```
Comment (theodesp) [inline docs/swagger.yaml:1]: With this yaml we can now generate  compatible Swagger clients

File: swagger.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
```
Comment (theodesp) [inline swagger.sh:1]: Helper script to re-generate the docs manually for now.

**Comment by prabhatsharma** [comment]
Swagger support is good. Thanks for the PR.

**Pull request merged**